hs_core/models.py

```diff
@@ -2025,26 +2523,20 @@ def get_crossref_deposit_xml(self, pretty_print=True):
 
     @property
     def size(self):
+        """
+        Return the total size of all data files in iRODS.
 
-        total_file_size = 0
-
+        This size does not include metadata. Just files. Specifically,
+        resourcemetadata.xml, systemmetadata.xml are not included in this
+        size estimate.
+        """
         try:
             # compute the total file size for the resource
-            f_sizes = [f.resource_file.size
-                       if f.resource_file else 0
-                       for f in self.files.all()]
-            total_file_size = sum(f_sizes)
-
-            # compute the total file size for federated resource
-            f_sizes = [int(f.fed_resource_file_size)
-                       if f.fed_resource_file_size else 0
-                       for f in self.files.all()]
-            total_file_size += sum(f_sizes)
+            f_sizes = [f.size for f in self.files.all()]
+            return sum(f_sizes)
 
         except SessionException:
-            pass
-
-        return total_file_size
+            return -1
```

Comment:
@alvacouch returning -1 when there is an exception may break user tracking code when federation is not available. They want to continue counting where there is an exception for some files which may also have some utility for other use cases, but I can see the value to just return -1 rather than giving false impression that the returned size is valid even though some files in the resource have some issues with returning the correct size.